js/ui/grid_core/ui.grid_core.keyboard_navigation.js

```diff
@@ -1603,7 +1604,8 @@ var KeyboardNavigationController = core.ViewController.inherit({
             columnIndex = that.getView("rowsView").getCellIndex($cellElement),
             rowIndex = this._getRowIndex($cellElement && $cellElement.parent()),
             isEditingCell = that.getController("editing").isEditCell(rowIndex, columnIndex),
-            row = dataController.items()[rowIndex];
+            localRowIndex = Math.min(rowIndex - dataController.getRowIndexOffset(), dataController.items().length - 1),
```

Comment:
Expression

rowIndex - dataController.getRowIndexOffset() +  dataController.getRowIndexDelta()

should be used to convert global row index to local. The getRowIndexDelta method call needs for rowRenderingMode virtual